core/trino-main/src/main/java/io/trino/operator/scalar/CharacterStringCasts.java

```diff
@@ -121,6 +121,24 @@ public static Slice varcharToCharSaturatedFloorCast(@LiteralParameter("y") long
         return codePointsToSliceUtf8(codePoints);
```

Comment:
could you add a test case in `io.trino.execution.TestCoordinatorDynamicFiltering#testJoinWithImplicitCoercion`?